lib/storage/client.js

```diff
@@ -1182,6 +1183,267 @@ class Client extends CryptoConsumer {
+    // Create the new membership key
+    // for loop
+    for (var i = 0; i < groupMembers.length; i++) {
+      let newMembershipKey = await this.crypto.createGroupMembershipKey(
+        this.config.privateKey,
+        groupMembershipJson.encrypted_membership_key,
+        publicKeysJSON.clients[groupMembers[i].clientID].public_key[
+          DefaultEncryptionKeyType
+        ],
+        publicKeysAuthJSON.public_key[DefaultEncryptionKeyType]
+      )
+      groupMembers[i].membershipKey = newMembershipKey
+    }
+    // Take the Given Paramaters and create the object needed for endpoint
+    let groupMembersToAdd = []
+    groupMembers.forEach(member => {
+      groupMembersToAdd.push({
+        client_ids: member.clientID,
+        membership_key: member.membershipKey,
+        capability_names: member.capabilities,
+      })
+    })
+    let addGroupMembersRequest = {
+      group_id: groupId,
+      group_members: groupMembersToAdd,
+    }
+    // Call the API endpoint to add group members
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/members`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(addGroupMembersRequest),
+      }
+    )
+    // Await the Response from the API with the Added Group member Response
+    let groupMembersJson = await validateResponseAsJSON(response)
+    let listGroupMembers = []
+    groupMembersJson.forEach(member => {
+      listGroupMembers.push(member)
+    })
+    return listGroupMembers
+  }
+
+  /**
+   * removeGroupMembers removes members to the TozStore Group given.
+   *
+   * @param {string} groupId  UUID assigned by TozStore, used to identify a group.
+   * @param {Array} clientIds all the clientIDs corresponding to the group members you want to remove
+   *
+   *
+   * @returns {Promise<bool>}
+   *
+   */
+  async removeGroupMembers(groupId, clientIds = []) {
+    // Take the Given Paramaters and create the object needed for endpoint
+    let groupMemberToRemove = []
+    clientIds.forEach(client => {
+      groupMemberToRemove.push({
+        client_id: client,
+      })
+    })
+    let removeGroupMembersRequest = {
+      group_id: groupId,
+      group_members: groupMemberToRemove,
+    }
+    // Call the API endpoint to remove group members
+    let response = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/members`,
+      {
+        method: 'DELETE',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(removeGroupMembersRequest),
+      }
+    )
+    // Check the response code
+    switch (response.status) {
+      case 204:
+        return true
+      case 403:
+        throw new Error('Unauthorized')
+      default:
+        throw new Error('Error while deleting record data.')
```

Comment:
Useful to add the status code to of response to provide some context of the failure, also we're "removing group members" here